Pik/fix pico neighbors

talbrecht opened the pull request:
This code adds 
1) PICO functionality in the case of ice shelves, that are confined within basins, that have no direct access to the open ocean. Those isolated basin regions are identified using the connected component method, but for each basin number individually, which may come with quite some cost?!
2) Also, PICO basin neighbors (and non-neighbors) are calculated only once at init per default, and not during every update any more (optional).
3) The lower bound for ocean input temperatures for PICO can be optionally set to pressure melting point, assuming that there is no process cooling ocean waters on their way to the grounding line.


**Checklist**

- [x] update documentation
- [x] update [`CHANGES.rst`](CHANGES.rst)
- [x] add regression tests

talbrecht committed:
Compute PICO basin neighbors only at init, only optional update every timestep

talbrecht committed:
Introduce isolated_masin_mask to find adjacent basin neighbors for ocean input. Could be merged with continental_shelf_mask?

talbrecht committed:
Added case in test/regression/pico_split

talbrecht committed:
Merge branch 'dev' of github.com:pism/pism into pik/fix_pico_neighbors

talbrecht committed:
Introduce option to bound PICO ocean temperatures by surface pressure melting

ckhroulev commented:
We need to save basin adjacency info in output files and read it in when re-starting.

ckhroulev committed:
Save basin adjacency info to output files

ckhroulev committed:
Rename ocean.pico.update_pico_basin_neighbors

(I removed "_pico" because it is redundant.)

ckhroulev committed:
Read basin neighbors from the input file when re-starting

- When bootstrapping (`-bootstrap` is set), the basin adjacency info is computed using the
  current ice extent (at the beginning of the run) and the basin mask read from
  `ocean.pico.file`.

- When re-starting (`-bootstrap` is not set), the basin mask and the basin adjacency info
  are read from the input file set using `-i input.nc`.

- Report basin neighbors (and where they came from) during initialization.

talbrecht committed:
Added some infos on changes in PICO basin neighbor treatment

talbrecht committed:
Added some infos on PICO basin neighbors and ocean temperature limit in manual, config snd changes.log

talbrecht committed:
Make sure that isolated pico mask is not overwritten when iterating over basins

Connected components finds all candidates with values 1 (on the continental shelf within basin b) connected to value 2 (open ocean in basin b). When iterating over basins b, isolated_pico_mask with value 1 can be found to be connected to other basin and overwritten. This has been fixed here...
